models/offer.go

```diff
@@ -38,6 +42,37 @@ const enabledOffers = `
 		AND offers.enabled = true
 `
 
+func buildScope(enabledOffers string, filterAttrs map[string]string) string {
+	subQueries := []string{enabledOffers}
+	for k, v := range filterAttrs {
+		// TODO: Possible SQL injection
```

Comment:
I just sent a commit creating a whitelist of valid characters and validating both at offer creation and offer query.